drivers/SmartThings/zigbee-fan/src/itm-fan-light/init.lua

```diff
@@ -0,0 +1,182 @@
+-- Copyright 2024 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+local stDevice = require "st.device"
+local clusters = require "st.zigbee.zcl.clusters"
+local capabilities = require "st.capabilities"
+local FanControl = clusters.FanControl
+local Level = clusters.Level
+local OnOff = clusters.OnOff
+
+local FINGERPRINTS = {
+  { mfr = "Samsung Electronics", model = "SAMSUNG-ITM-Z-003" },
+}
+
+local function can_handle_itm_fanlight(opts, driver, device, ...)
+  for _, fingerprint in ipairs(FINGERPRINTS) do
+    if device:get_manufacturer() == fingerprint.mfr and device:get_model() == fingerprint.model then
+      return true
+    end
+  end
+  return false
+end
+
+local levels_for_speed = {
+  [0] = 0,
+  [1] = 25,
+  [2] = 50,
+  [3] = 100,
+}
+
+local function level_to_speed(level)
+  local speed = 0
+  if level == 0 then
+    speed  = 0
+  elseif level  > 0 and level <= 25 then
+    speed = 1
+  elseif level > 25 and level <= 75 then
+    speed = 2
+  else
+    speed = 3
+  end
+  return speed
+end
+
+-- CREATE CHILD DEVICE
+
+local function create_child_devices(driver, device, profile, child_type)
+  local metadata = {
+    type = "EDGE_CHILD",
+    parent_assigned_child_key = child_type,
+    label = string.format("%s %s", device.label, child_type),
+    profile = profile,
+    parent_device_id = device.id,
+    vendor_provided_label = string.format("%s %s", device.label, child_type)
+  }
+  driver:try_create_device(metadata)
+  device:refresh()
+end
+
+local function device_added(driver, device)
+  if device.network_type ~= stDevice.NETWORK_TYPE_CHILD then
+    create_child_devices(driver, device, 'switch-level', 'light')
+  end
+end
+
+local function device_init(driver, device)
+  local dev = device:get_parent_device()
+    if dev == nil then
+    dev = device
+  end
+     dev:refresh(driver, dev)
```

Comment:
nit:spacing

additionally, there are race conditions involved when you call `refresh` from `init`, as the zigbee stack may not be fully initialized.